src/zenml/zen_stores/schemas/pipeline_run_schemas.py

```diff
@@ -52,6 +53,13 @@ class PipelineRunSchema(NamedSchema, table=True):
     """SQL Model for pipeline runs."""
 
     __tablename__ = "pipeline_run"
+    __table_args__ = (
+        UniqueConstraint(
+            "orchestrator_run_id",
+            "deployment_id",
+            name="unique_orchestrator_run_id_for_deployment",
+        ),
+    )
```

Comment:
Is a new deployment created every time you do `python run.py`? Or is there any scenario other than scheduling where deployments are shared among runs?

I'm asking because placeholder runs are now created with empty `orchestrator_run_id` which would fail if there was any other placeholder run with the same deployment ID.

Reply:
Yes, that's correct. The only place where we create deployments and placeholder runs is in the `Pipeline._run` method which means there is a 1-1 relationship between those.